File: sampler.java
import com.synthbot.audioplugin.vst.JVSTPlugin;
import com.synthbot.audioplugin.vst.JVSTPluginPrograms;

public class Sampler extends JVSTPlugin {
  private float[] samples;
  private float pitch;

  public Sampler(long wrapper) {
    super(wrapper);
  }

  @Override
  public void processAudio(float[][] inputs, float[][] outputs) {
    float[] inL = inputs[0];
    float[] inR = inputs[1];
    float[] outL = outputs[0];
    float[] outR = outputs[1];

    for (int i = 0; i < outL.length; i++) {
      int sampleIndex = (int) (i * pitch);
      float sample = samples[sampleIndex % samples.length];
      outL[i] = sample;
      outR[i] = sample;
    }
  }

  @Override
  public float getParameter(int index) {
    switch (index) {
      case 0:
        return pitch;
      default:
        return 0;
    }
  }

  @Override
  public void setParameter(int index, float value) {
    switch (index) {
      case 0:
        pitch = value;
        break;
    }
  }

  @Override
  public int getNumPrograms() {
    return 1;
  }

  @Override
  public String getProgramName(int index) {
    return "Sampler";
  }

  @Override
  public void setProgram(int index) {
  }

  @Override
  public JVSTPluginPrograms getPrograms() {
    return null;
  }
}

File: sampler.py
import pyvst

class Sampler(pyvst.Plugin):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

        # Create a list to store the samples
        self.samples = []

        # Create a parameter for the pitch of the sample
        self.pitch_param = pyvst.FloatParam(0.5, min=0.0, max=1.0, default=0.5, displayName="Pitch")
        self.add_param(self.pitch_param)

    def get_parameter(self, index):
        if index == 0:
            return self.pitch_param.value
        else:
            return 0.0

    def set_parameter(self, index, value):
        if index == 0:
            self.pitch_param.value = value

    def process_audio(self, inputs, outputs):
        # Get the audio input and output buffers
        in_l, in_r = inputs
        out_l, out_r = outputs

        # Process the audio samples
        for i in range(len(out_l)):
            # Calculate the pitch-shifted sample index
            sample_index = int(i * self.pitch_param.value)

            # Get the sample from the list of samples
            sample = self.samples[sample_index % len(self.samples)]

            # Write the sample to the output buffers
            out_l[i] = sample
            out_r[i] = sample
